implement debounce with dummy json

colespen opened the pull request:
made simple Search component update using debounce hook and a search helper which sorts fake JSON.

jasoncolburne commented:
Hey thanks Cole! You did way more than I expected, I should have been clearer, I was talking about the search field on the admin user's page, I had made my own debounce for that. This though, the search functionality, it's going to be great when we hook that up. I'll leave some more comments on the PR itself when I have time tonight.

jasoncolburne commented:
Notice you checked in the apple DS_Store files, can we actually add those to .gitignore?

jasoncolburne commented:
This is the code I did mean: https://github.com/jasoncolburne/arad/blob/main/src/front-end/src/identification/Users/components/UserList.tsx#L26-L38

colespen commented:
> This is the code I did mean: https://github.com/jasoncolburne/arad/blob/main/src/front-end/src/identification/Users/components/UserList.tsx#L26-L38

Hey @jasoncolburne, I implemented this basic debounce you originally wanted. FYI I just replaced User with MockUser interface to for development.

jasoncolburne commented on src/front-end/src/identification/Users/components/UserList.tsx:
```diff
@@ -19,23 +26,21 @@ interface UserListProps {
-      setCurrentTimer(null);
-    }, 666);
+    },
+    575
```
Why 575, is this just recommended?

colespen replied:
I was experimenting with different durations and I arbitrarily left it with 575.

jasoncolburne commented:
Hey @colespen can you change the `MockUser` stuff back into User so I can run CI against this? Generally you wouldn't push that kind of local change to github.
Thanks!

colespen commented:
> Hey @colespen can you change the `MockUser` stuff back into User so I can run CI against this? Generally you wouldn't push that kind of local change to github. Thanks!

Hey @jasoncolburne. Ok I understand. Done.

jasoncolburne commented on src/front-end/src/GlobalState.tsx:
```diff
@@ -64,6 +64,7 @@ const modifyAccessToken = (
         },
       },
       user: state.user!,
+      query: state.query!
```
Are we sure the global state is the right place for this? Can it be somehow bound to the search page? It really only applies there

colespen replied:
Ok, I am working on this: 
Perhaps it would be another job for useContext()? Currently I have the SearchBar component within Header/Navigation components which is not ideal for passing down props. And yes, ultimately, the state change  occurs in SearchBar.

jasoncolburne commented on src/front-end/src/components/Footer/index.tsx:
```diff
@@ -48,7 +49,7 @@ const Footer = () => {
       <div className="footer">
         <div className="left">
-          <Link id="arad-codeLink" to="/code">
+          <Link id="arad-codeLink" to={location.pathname !== "/search" ? "/code" : "/"}>
```
So this will change the link in the lower right, correct? Meaning that if the path is /search it will go to the code, but if ti's not it will redirect home? Why are we doing this? Can't we alway find home through another link?

colespen replied:
I did it so I could use the arad title (bottom left) as a link back to home, in this case to exit the search and return. Yes I can totally remove it if we want a different nav button for home.

jasoncolburne commented on src/front-end/src/components/Header/index.tsx:
```diff
@@ -33,7 +33,7 @@ const Header = () => {
   React.useEffect(() => {
     const encoded_state: string | null = localStorage.getItem("state");
-    if (encoded_state) {
+    if (encoded_state && encoded_state !== "undefined") {
```
Why do we need this guard?

colespen replied:
without it I was seeing this in the console:
```bash
"undefined" is not valid JSON
```

jasoncolburne commented on src/front-end/src/components/Header/components/Navigation.tsx:
```diff
@@ -18,11 +22,16 @@ const adminLinks = (
     <header>
       <nav>
-        <Link to="/search">Search</Link>
+        {location.pathname !== "/search" ? (
```
Maybe we could just pass a prop down to SearchBar and have it render the right thing rather than doing the conditional up here? That way it's always just a SearchBar, and the behaviour can live in the component.

jasoncolburne commented on src/front-end/src/helper/searchDatabase.ts:
This file is temporary, correct? You needed it because you didn't have a backend?

colespen replied:
correct!

jasoncolburne commented on src/front-end/src/hooks/useDebounce.ts:
```diff
@@ -0,0 +1,12 @@
+  }, [input, ms]);
+
+  return debounced;
+}
```
In general can you go through and add newlines to these files where one is missing at the end?

jasoncolburne commented on src/front-end/src/identification/Users/components/UserListRow.tsx:
```diff
@@ -7,12 +7,15 @@ import {
 interface UserListRowProps {
-  user: User;
+  // ** User
+  user: MockUser;
```
I think if I run CI these changes will break tests.

colespen replied:
I will definitely fix this first.
